packages/browser-crawler/src/internals/browser-crawler.ts

```diff
@@ -715,7 +715,7 @@ export async function browserCrawlerEnqueueLinks({
  */
 // eslint-disable-next-line @typescript-eslint/ban-types
 async function extractUrlsFromPage(page: { $$eval: Function }, selector: string, baseUrl?: string): Promise<string[]> {
-    const urls = await page.$$eval(selector, (linkEls: HTMLLinkElement[]) => linkEls.map((link) => link.getAttribute('href')).filter((href) => !!href)) ?? [];
+    const urls = await page.$$eval(selector, (linkEls: HTMLLinkElement[]) => linkEls.map((link) => link.href).filter((href) => !!href)) ?? [];
 
     return urls.map((href: string) => {
         // Throw a meaningful error when only a relative URL would be extracted instead of waiting for the Request to fail later.
```

Comment:
I guess the following part could be deleted as well, because the `.href` should always return an absolute URL. At least I think so and have never seen it do otherwise 😅

Reply:
Yeah just confirmed, it always gives you absolute URLs, even if the `href` is empty or a fragment only.